### DISCUSSION TITLE
How to reference a set of Strings

### DISCUSSION THREAD

**Discussion by paulfrench** [eclipse-langium/langium#322] [General]
I'm trying to convert some Xtext to Langium, I've got several problems but let's deal with one at a time..

`Application:
	'application' '{'
	    'name' ':' name=STRING
	    'target-platform' ':' targetPlatform=TargetPlatform
	'}';`

What do I set TargetPlatform to? In XText I could use an enum...

`enum TargetPlatform:
	Azure | AWS | local;`

enum does not seem to be supported.

Thanks

`

**Comment by msujew** [comment]
Hi @paulfrench,

we removed Enums in favor of using data type rules. They were already interchangeable in Xtext and due to some additional generated types, we didn't see the need to have them in Langium.

A corresponding data type rule would look like this:
```
TargetPlatform returns string: 'Azure' | 'AWS' | 'local';
```

This generates the `TargetPlatform` type in the `ast.ts`:
```ts
type TargetPlatform = 'Azure' | 'AWS' | 'local';
```

**Reply by paulfrench** [reply]
Thanks, my other issues went away when I used the create syntax. Cheers
